zap/src/main/java/org/zaproxy/zap/authentication/PostBasedAuthenticationMethodType.java

```diff
@@ -377,8 +380,10 @@ public WebSession authenticate(
             // Add message to history
             AuthenticationHelper.addAuthMessageToHistory(msg);
 
-            user.getAuthenticationState()
-                    .setLastAuthRequestHistoryId(msg.getHistoryRef().getHistoryId());
+            if (msg.getHistoryRef() != null) {
+                user.getAuthenticationState()
+                        .setLastAuthRequestHistoryId(msg.getHistoryRef().getHistoryId());
+            }
```

Comment:
Provide the stack traces of the two exceptions.

Reply:
Are you thinking something like this instead?

```
try {
    user.getAuthenticationState()
            .setLastAuthRequestHistoryId(msg.getHistoryRef().getHistoryId());
}catch (Exception e) {
    LOGGER.error("Unable to set last auth request history: " + e.getMessage(), e);
}
```

Reply:
Sorry, was asking about the stack traces you saw in the logs not for changes in the code. Just curiosity.

Reply:
Ah yes, I added this check because when testing/mocking I wasn't able to add a history ref for the `msg` and the line above `AuthenticationHelper.addAuthMessageToHistory(msg)` simply logs if a history ref was not able to be saved, so it seemed appropriate to do the same here for a missing ref.

Reply:
I thought you saw this in production. Ok, not ideal changing it for the tests but the NPE could still happen so.